addons/medical_ai/functions/fnc_healUnit.sqf

```diff
@@ -27,7 +27,13 @@ private _healQueue = _this getVariable [QGVAR(healQueue), []];
 private _target = _healQueue param [0, objNull];
 
 // If unit died or was healed, be lazy and wait for the next tick
-if (!alive _target || {!(_target call FUNC(isInjured))}) exitWith {
+// If the unit can't be healed, go to the next unit to be healed
+if (!alive _target || {!(_target call FUNC(isInjured))} || {GVAR(requireItems) != 0 && {
+    private _treatmentEvent = (_this getVariable [QGVAR(currentTreatment), []]) param [2, ""];
+
+    // Target still needs healing, but the healer doesn't have the required items or needs to wait
+    _treatmentEvent in ["#needsBandages", "#needsBandagesOrTourniquets", "#needsIV", "#needsFewerMeds", "#needsSlowerHeart", "#needsFasterHeart"]
```

Comment:
just a possible change, not sure if it's any better

```suggestion
    (_treatmentEvent select [0, 6]) == "#needs"
```

Reply:
It is. I'm adding a `#needsSplint` state, and having your code would make the check easier overall.